src/VecSim/algorithms/brute_force/brute_force_single.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "brute_force.h"
+
+class BruteForceIndex_Single : public BruteForceIndex {
+private:
+    vecsim_stl::unordered_map<labelType, idType> labelToIdLookup;
+
+public:
+    BruteForceIndex_Single(const BFParams *params, std::shared_ptr<VecSimAllocator> allocator);
+    ~BruteForceIndex_Single();
+
+    virtual int addVector(const void *vector_data, size_t label) override;
+    virtual int deleteVector(size_t id) override;
+    virtual double getDistanceFrom(size_t label, const void *vector_data) const override;
+
+    virtual inline size_t indexLabelCount() const override { return this->count; }
+
+private:
```

Comment:
If those functions are private, you have a duplicated private section. Also, why not protected?

Reply:
Because no one inherits this class, they can be private.
Is it a problem to have duplicate sections? I think it is more readable to have the members first, then the public functions, and in the end, the private ones.